File: benchmarks/tsb/bench_series_transform.ts
/**
 * Benchmark: seriesTransform on 100k-element Series
 */
import { Series, seriesTransform } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const data = Array.from({ length: ROWS }, (_, i) => i * 0.1);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) seriesTransform(s, (v) => (v as number) ** 2);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) seriesTransform(s, (v) => (v as number) ** 2);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "series_transform",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_transform.py
"""Benchmark: Series.transform on 100k-element pandas Series"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
s = pd.Series([i * 0.1 for i in range(ROWS)])

for _ in range(WARMUP):
    s.transform(lambda v: v ** 2)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.transform(lambda v: v ** 2)
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "series_transform", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
